src/aiida_quantumespresso/parsers/dos.py

```diff
@@ -3,46 +3,36 @@
 import numpy as np
 
 from aiida_quantumespresso.parsers import QEOutputParsingError
-from aiida_quantumespresso.parsers.parse_raw.base import parse_output_base
+from aiida_quantumespresso.utils.mapping import get_logging_container
 
-from .base import Parser
+from .base import BaseParser
 
 
-class DosParser(Parser):
-    """This class is the implementation of the Parser class for Dos."""
+class DosParser(BaseParser):
+    """``Parser`` implementation for the ``DosCalculation`` calculation job class."""
 
     def parse(self, **kwargs):
-        """Parses the datafolder, stores results.
+        """Parse the retrieved files of a ``DosCalculation`` into output nodes."""
+        logs = get_logging_container()
 
-        Retrieves dos output, and some basic information from the out_file, such as warnings and wall_time
-        """
-        retrieved = self.retrieved
+        _, parsed_stdout, logs = self.parse_stdout_from_retrieved(logs)
 
-        # Read standard out
-        try:
-            filename_stdout = self.node.get_option('output_filename')  # or get_attribute(), but this is clearer
-            with retrieved.base.repository.open(filename_stdout, 'r') as fil:
-                out_file = fil.readlines()
-        except OSError:
-            return self.exit(self.exit_codes.ERROR_OUTPUT_STDOUT_READ)
-
-        job_done = False
-        for i in range(len(out_file)):
-            line = out_file[-i]
-            if 'JOB DONE' in line:
-                job_done = True
-                break
-        if not job_done:
-            return self.exit(self.exit_codes.ERROR_OUTPUT_STDOUT_INCOMPLETE)
-
-        # check that the dos file is present, if it is, read it
+        base_exit_code = self.check_base_errors(logs)
+        if base_exit_code:
+            return self.exit(base_exit_code, logs)
+
+        self.out('output_parameters', Dict(parsed_stdout))
+
+        for exit_code in ['ERROR_OUTPUT_STDOUT_INCOMPLETE']:
+            if exit_code in logs.error:
+                return self.exit(self.exit_codes.get(exit_code), logs)
```

Comment:
I saw how you used this for/if construct in `check_base_errors` but here it makes little sense since there is just one exit code you check for and there is no other loop where there are multiple to keep the symmetry with. So here I see no reason to keep the loop. 
```suggestion
        if 'ERROR_OUTPUT_STDOUT_INCOMPLETE'in logs.error:
            return self.exit(self.exit_codes.ERROR_OUTPUT_STDOUT_INCOMPLETE, logs)
```
idem for other parsers

Reply:
Hehe, fair. I guess the idea was that a future developer would add more of these later and the `for/if` constructor would make it obvious on how to add them. But those future developers will most likely be one of us and it's not that complicated anyway.

Reply:
Actually, it should probably be this:

```
        for exit_code in list(self.get_error_map().values()) + ['ERROR_OUTPUT_STDOUT_INCOMPLETE']:
            if exit_code in logs.error:
                return self.exit(self.exit_codes.get(exit_code), logs)
```

Though currently the only error in there is the `ERROR_OUT_OF_WALLTIME`. Which also isn't defined on every `NamelistsCalculation` subclass. Will go through the various parsers and see where it makes sense to add.

Reply:
See [7460b6c](https://github.com/aiidateam/aiida-quantumespresso/pull/756/commits/7460b6cf5e46dc48a0c536175305d898b5fbf22d) and [064df48](https://github.com/aiidateam/aiida-quantumespresso/pull/756/commits/064df48326e58f1a443530a4e89f2b8f5f18f78c)